pytorch_lightning/utilities/model_summary.py

```diff
@@ -310,9 +312,9 @@ def _get_summary_data(self):
             arrays.append(["In sizes", self.in_sizes])
             arrays.append(["Out sizes", self.out_sizes])
 
-        return arrays
+        return arrays  # type: ignore
```

Comment:
`pytorch_lightning/utilities/model_summary.py:315: error: Incompatible return value type (got "List[List[Sequence[str]]]", expected "List[List[Union[str, List[str]]]]")`

why???

Reply:
I'm not exactly sure about why it converts it to `Sequence`, however, the return annotation doesn't seem right to me.

`List[Union[str, List[str]]]` suggests that it can be of any length and that `str` and `List[str]` can be in any order, however, the actual format represents just two elements where the first is a `str` and the second a `List[str]`.

For that reason, I'd annotate this return type as `List[Tuple[str, List[str]]]` and update the code to this stricter requirement:

```python
        arrays = [
            (" ", [str(i) for i in range(len(self._layer_summary))]),
            ("Name", self.layer_names),
            ("Type", self.layer_types),
            ("Params", [get_human_readable_count(n) for n in self.param_nums]),
        ]
        if self._model.example_input_array is not None:
            arrays.append(("In sizes", self.in_sizes))
            arrays.append(("Out sizes", self.out_sizes))
        return arrays
```

note: list comprehensions are more pythonic than `list(map(str` :D

Reply:
great, thanks that works. I won't change the rest of the code, I want this PR to be just about typing.